src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utilities for testing COS integration with charms."""
+import logging
+from pathlib import Path
+from typing import Any, Dict, Set
+
+import yaml
+from juju.action import Action
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+
+log = logging.getLogger(__name__)
+
+GRAFANA_AGENT_APP = "grafana-agent-k8s"
+METRICS_ENDPOINT_RELATION = "metrics-endpoint"
+GRAFANA_DASHBOARD_RELATION = "grafana-dashboard"
+ALER_RULES_DIRECTORY = Path("./src/prometheus_alert_rules")
+
+
+async def deploy_and_assert_grafana_agent(
+    model: Model,
+    app: str,
+    channel: str = "latest/stable",
+    dashboard: bool = True,
+    metrics: bool = True,
+) -> None:
+    """Deploy grafana-agent-k8 and add relate it with app.
+
+    Helper function to deploy and relate grafana-agent-k8s with provided app.
+
+    Args:
```

Comment:
Not sure, I'm not familiar with this type of docstring, let me search for some good practice. 

Most of the time I'm not using any, since it can be generated from type hinting or using this format:
```python
:param app: Application name.
:type app: str
```

After reading some tutorial like this [one](https://www.dataquest.io/blog/documenting-in-python-with-docstrings/), we are using the  Google docstrings and it should be like this:
```python
app (str): name of application with which the Grafana agent should be related
```
Is that ok?
```